nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,68 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    console.log(meals);
+    console.log("in /api/meals");
+    // to verify for empty object
+  if (
+    Object.keys(request.query).length === 0 &&
+    request.query.constructor === Object
+  ) {
+    response.send(meals);
+  } else if (request.query.maxPrice) {
+    if (isNaN(Number(request.query.maxPrice))) {
+      response.status(400).send("Invalid id");
+    } else {
+      response.send(
+        meals.filter((meal) => {
+          return meal.price < Number(request.query.maxPrice);
+        })
+      );
+    }
+  } else if (request.query.title) {
+    response.send(
+      meals.filter((meal) => meal.title.toLowerCase().includes(`${request.query.title.toLowerCase()}`))
+    );
+  } else if (request.query.createdAfter) {
+    if (isNaN(Date.parse(request.query.createdAfter))) {
+      response.status(400).send(" date not parsable");
+    } else {
+      response.send(
+        meals.filter(
+          (meal) =>
+            Date.parse(meal.createdAt) >
+            Date.parse(request.query.createdAfter)
+        )
+      );
+    }
+  } else if (request.query.limit) {
+    if (isNaN(Number(request.query.limit))) {
+      response.status(400).send("Invalid id");
+    } else {
+      const limit = Number(request.query.limit);
+      const limitedMeals = [];
+      for (let i = 0; i < limit; i++) {
+        limitedMeals.push(meals[i]);
+      }
+      response.send(limitedMeals);
```

Comment:
All the query parameters work individually, what if there are more than one query?
For ex: both maxPrice and limit should apply to meals.